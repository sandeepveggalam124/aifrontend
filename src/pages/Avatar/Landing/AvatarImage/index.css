.embed-container{
    background-color: black;
    display: flex;
    justify-content: center;
    height:100%;
    align-items: end;
    position:relative;
}
.embed-container img{
    max-width: 100%;
    max-height:100%;
    /* object-fit: cover; */
    /* width:auto;
    height:auto; */
    /* position:absolute;
    bottom:0 */
}