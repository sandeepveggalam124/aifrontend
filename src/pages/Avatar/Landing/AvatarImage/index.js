import React,{memo} from 'react'
import './index.css'


const ImageComponent=memo(({src,alt})=>{
  return(
  
      <div className='embed-container p-0'>
        <img src={src} alt={alt} loading='lazy'></img>

      </div>
  )
})
export default ImageComponent