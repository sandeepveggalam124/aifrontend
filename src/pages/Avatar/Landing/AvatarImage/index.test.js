import { MemoryRouter as Router } from "react-router-dom";
import {render,screen} from "@testing-library/react";
import AvatarImage from "./index";

describe('Image Component',()=>{
it('renders correctly',()=>{
    const {container}=render(
        <Router>
            <AvatarImage/>
        </Router>
    )
    expect(container).toMatchSnapshot()
})
test('renders with correct props',()=>{
    const src='/assests/images/avatar.svg'
    const alt='Avatar Image'
    render(<AvatarImage src={src} alt={alt}/>)
    const imageElement=screen.getByAltText(alt)
    expect(imageElement).toBeInTheDocument();
    expect(imageElement).toHaveAttribute('src',src);


})
})