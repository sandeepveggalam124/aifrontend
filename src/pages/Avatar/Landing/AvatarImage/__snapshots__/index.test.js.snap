// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component renders correctly 1`] = `
<div>
  <div
    class="embed-container p-0"
  >
    <img
      loading="lazy"
    />
  </div>
</div>
`;
